From 9b8cce3228ee8a2bd0ec5d782fa1314139b5bbfc Mon Sep 17 00:00:00 2001
From: Yaoming Qin <b10945@freescale.com>
Date: Tue, 25 Jan 2011 16:01:07 +0800
Subject: [PATCH 115/486] ENGR00138392 A strip of screen flashes when switch
 to camcorder mode.

The root cause is that the input buffer of the v4l2 output is
not reset at the beginning.

Signed-off-by: Yaoming Qin<B10945@freescale.com>
---
 liboverlay/overlay.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/liboverlay/overlay.cpp b/liboverlay/overlay.cpp
index 21c28c9..303bff1 100755
--- a/liboverlay/overlay.cpp
+++ b/liboverlay/overlay.cpp
@@ -839,6 +839,12 @@ static int overlay_init_v4l(struct overlay_control_context_t *dev)
                 return -1;
             }
 
+			//reset the v4l input buffer
+			for (int i = 0; i < (dev->xres * dev->yres); i ++){
+				vir_addr[2*i] = 0;
+				vir_addr[2*i + 1] = 0x80;
+				}
+
             dev->v4lbuf_addr[buf_index] = vir_addr;
             //v4l already init this buffer to black
             //fill the v4l to black;
-- 
1.8.0

